### DISCUSSION TITLE
docker-clickhouse-1 using a lot of CPU

### DISCUSSION THREAD

**Discussion by dalcorsogabriel** [akvorado/akvorado#1268] [Q&A]
This service is using a lot of CPU and i wanted to know how to lower the usage... What can I change in akvorado.yaml and inlet.yaml to do it? I tried changing the numbers of workers and consumers in those, but didnt see any improving. Thanks.
![Clickhouse](https://github.com/akvorado/akvorado/assets/140973597/bf928dcc-0326-402a-a0ee-5bb4b7b9eb55)

**Comment by vincentbernat** [comment]
ClickHouse is the database. How many flows per second do you have in your setup?

**Reply by dalcorsogabriel** [reply]
![image](https://github.com/akvorado/akvorado/assets/140973597/4ebf2599-6b49-42e6-bcd2-9724bf5f8471)
I have 2k flows/s in the akvorado because i put a rate limit of 2k in inlet.yaml, but with 4 exporters i am sending 8k flows/s total

**Reply by vincentbernat** [reply]
The ClickHouse CPU is a bit high for 2000 flows/s. You can tune a bit the flush interval for ClickHouse. In `akvorado.yaml`:
```yaml
clickhouse:
  kafka:
    engine-settings:
      - kafka_flush_interval_ms = 1000
```
(merge that with your current configuration)

You'll need to restart the orchestrator to take effect.

**Reply by dalcorsogabriel** [reply]
Thanks for the tip. I apllied this config and is much better, around 40% of the CPU. Sometimes it still goes up to 200% and the docker-kafka too, but is much more stable.
I am using 16vCPUs anda 64Gb RAM. Around 2,6k flows/s. If you have any other suggestions of the workers and consumers configurations is much appreciated.
Just to have some parameters here are the configs apllied now:
akvorado.yaml:
clickhouse:
  orchestrator-url: http://akvorado-orchestrator:8080/
  kafka:
    consumers: 6
    engine-settings:
      - kafka_flush_interval_ms = 10000
  resolutions:
    - interval: 0
      ttl: 1h  # 1 hour
    - interval: 1m
      ttl: 2160h  # 3 month
    - interval: 5m
      ttl: 2160h # 3 months
    - interval: 1h
      ttl: 8760h # 1 year
      



inlet.yaml:
kafka:
  compression-codec: zstd
metadata:
  workers: 10
  provider:
    type: snmp
    
flow:
  inputs:
    - type: udp
      decoder: netflow
      listen: :9995
      workers: 10
      receive-buffer: 20971520
  rate-limit: 2000
core:
  workers: 10
  default-sampling-rate: 2000
  
 

Thanks for the help already Vincent.
